### PR TITLE
xtensa-build-zephyr.py: p ristine deletes STAGING_DIR

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits, see commit messages

### CODE REVIEW

**Review by dbaluta — APPROVED** [review]

File: scripts/xtensa-build-zephyr.py [hunk 1]
```diff
@@ -497,10 +497,12 @@ def build_platforms():
...
 		try:
 			execute_command(build_cmd, cwd=west_top)
-		except:
+		except subprocess.CalledProcessError as cpe:
 			zephyr_path = pathlib.Path(west_top, "zephyr")
 			if not os.path.exists(zephyr_path):
-				sys.exit("Zephyr project not found. Please run this script with -c flag or clone manually.")
+				sys.exit("Zephyr project not found. Please run this script with -u flag or `west update zephyr` manually.")
+			else: # unknown failure
+				raise cpe
```
Comment (kv2019i) [inline scripts/xtensa-build-zephyr.py:505]: Why hide other exceptions. I prefer https://github.com/thesofproject/sof/pull/6097

Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py:505]: Other exceptions are not hidden, they're not caught and raised directly

Comment (marc-hb) [reply scripts/xtensa-build-zephyr.py:505]: This fixes `pylint W0703: Catching too general exception Exception (broad-except)`

Comment (kv2019i) [reply scripts/xtensa-build-zephyr.py:505]: Fair enough, the exception catching is filtered now as well, right.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**